quickwit/quickwit-serve/src/index_api/rest_handler.rs

```diff
@@ -141,6 +145,91 @@ fn get_indexes_metadatas_handler(
         .map(make_response)
 }
 
+/// Describes an index with its main information and statistics.
+#[derive(Serialize, Deserialize, utoipa::ToSchema)]
+struct IndexStats {
+    pub index_id: String,
+    pub index_uri: Uri,
+    pub num_published_splits: usize,
+    pub num_published_docs: u64,
+    pub size_published_docs: Byte,
+    pub timestamp_field_name: Option<String>,
+    pub min_timestamp: Option<i64>,
+    pub max_timestamp: Option<i64>,
+}
+
+#[utoipa::path(
+    get,
+    tag = "Indexes",
+    path = "/indexes/{index_id}/describe",
+    responses(
+        (status = 200, description = "Successfully fetched stats about Index.", body = IndexStats)
+    ),
+    params(
+        ("index_id" = String, Path, description = "The index ID to describe."),
+    )
+)]
+
+/// Describes an index.
+async fn describe_index(
+    index_id: String,
+    metastore: Arc<dyn Metastore>,
+) -> Result<IndexStats, MetastoreError> {
+    let index_metadata = metastore.index_metadata(&index_id).await?;
+    let query = ListSplitsQuery::for_index(&index_id);
+    let splits = metastore.list_splits(query).await?;
+    let published_splits: Vec<Split> = splits
+        .into_iter()
+        .filter(|split| split.split_state == SplitState::Published)
+        .collect();
+    let mut total_num_docs = 0;
+    let mut total_bytes = 0;
+    let mut min_timestamp: Option<i64> = None;
+    let mut max_timestamp: Option<i64> = None;
+
+    for split in &published_splits {
+        total_num_docs += split.split_metadata.num_docs as u64;
+        total_bytes += split.split_metadata.footer_offsets.end;
+
+        for time_range in &split.split_metadata.time_range {
+            let start_time = *time_range.start();
+            if min_timestamp.is_none() || start_time < min_timestamp.unwrap() {
+                min_timestamp = Some(start_time);
+            }
+
+            let end_time = *time_range.end();
+            if max_timestamp.is_none() || end_time > max_timestamp.unwrap() {
+                max_timestamp = Some(end_time);
+            }
+        }
+    }
+
+    let index_config = index_metadata.into_index_config();
+    let index_stats = IndexStats {
+        index_id,
+        index_uri: index_config.index_uri.clone(),
+        num_published_splits: published_splits.len(),
+        num_published_docs: total_num_docs,
+        size_published_docs: Byte::from(total_bytes),
+        timestamp_field_name: index_config.doc_mapping.timestamp_field,
+        min_timestamp,
+        max_timestamp,
+    };
+
+    Ok(index_stats)
+}
+
+fn describe_index_handler(
+    metastore: Arc<dyn Metastore>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    warp::path!("indexes" / String / "describe")
+        .and(warp::get())
+        .and(with_arg(metastore))
+        .then(describe_index)
+        .and(extract_format_from_qs())
```

Comment:
you don't need `extract_format_from_qs`, it would be nice to have the pretty json query param.

Reply:
if i am going to remove `extract_format_from_qs`, what should i use then? can you please elaborate more ?

Reply:
Oh sorry, my mistake, this is the right one…

Reply:
ok :)